### ISSUE TITLE
Test NetworkServiceMesh with k8s

### ISSUE THREAD

**Issue by daviddias** [ipfs/testground#298]
https://networkservicemesh.io/

**Comment by raulk** [comment]
@nonsense mind providing a status update here? I believe we've made our CNI choices and integrated them with Sidecar, but it would be great if you could summarise your findings here if we ended up doing a spike on NetworkServiceMesh, or the rationale why we dropped it, if we didn't do a spike. Thanks!

**Comment by nonsense** [comment]
I reviewed https://networkservicemesh.io/ and also spoke with one of the maintainers - in theory it should be possible to use it for our `data` plane, because it works as ortogonal set of networking services, alongside the default Kubernetes network.

However I had problems installing it and testing it - I tried with `kops` and `flannel` and it wouldn't boot up - I also tried with `kind` and again it was crashing on start-up.

This is not unusual in the k8s world, because there are multiple ways of setting up a cluster (I had similar problems with the other CNIs, usually you have to adjust a config or two).

Considering that I couldn't get the basic functionality up and running on `kops` and `kind` and I had hard time finding any architecture docs, apart from the examples, that don't map very well to our specific problem, I stopped at this point.

It is not clear what kind of traffic shaping functionality we have out of the box (if any) in NSM.

Bottom line - I think we should not discard this, and revisit in case we discover any show-stoppers with Weave.

**Comment by raulk** [comment]
Closing this issue. We'll reopen and revisit this if our current setup weave + flannel + CNI Genie fails us in some manner.
